src/App.tsx

```diff
@@ -1,36 +1,192 @@
-import React from 'react';
+import React, {useState, useEffect, useContext, Dispatch, SetStateAction} from 'react';
+import {BrowserRouter as Router, Routes, Route, Link} from 'react-router-dom';
 import logo from './logo.svg';
 import nasaLogo from './NASA.svg';
 import './App.css';
 
-function MyTemplate(props: any){
+interface CounterProps {
+  counter: {
+    count: number;
+    setCount: Dispatch<SetStateAction<number>>;
+  }
+}
+
+interface MyTemplateProps {
+  name: string;
+  image: string;
+  children: JSX.Element[];
+}
+
+function Count() {
+  let countText = localStorage.getItem('count');
+  let countFromLocalStorage = countText ? parseInt(countText!) : 0;
+
+  const [count, setCount] = useState(countFromLocalStorage);
+
+  useEffect(() => {
+    localStorage.setItem('count', count.toString());
+  });
+
+  return (
+      <div>
+        <p>You clicked {count} times</p>
+        <button onClick={() => setCount(count + 1)}>
+          Click me
+        </button>
+      </div>
+  );
+}
+
+
+function Component1(){
+  let countText = localStorage.getItem('count2');
+  let countFromLocalStorage = countText ? parseInt(countText!) : 0;
+  const [count, setCount] = useState(countFromLocalStorage);
+  useEffect(() => {
+    localStorage.setItem('count2', count.toString());
+  });
```

Comment:
you could add a custom hook for this as it's used in 2 components

take a look at https://reactjs.org/docs/hooks-custom.html